### DISCUSSION TITLE
Query IDs that have been encoded with pg_hashids?

### DISCUSSION THREAD

**Discussion by taeling** [supabase/supabase#30980] [Questions] [answered]
I could use a pointer...

I am trying to use id_decode and id_encode from pg_hashids - but I haven't found a way to decode an id when trying to select or update a specific row using the sdk.

### 1. Table with bigint id
`create table messages (
   id bigint primary key generated always as identity,
   message text
);`

## 2. View with encoded id
```
create view encoded_messages as 
select
  id_encode(id),
  message
from 
  messages;
```
## 3. Query encoded view

`supabase.from('encoded_messages').select('*')`

let's say that returns the id '_jJPz_' .

## 4. Update message based on encoded id

`supabase.from("messages").update(...).eq("id","id_decode_once('jJPz')")`

The error I am getting back is 

{
    "code": "22P02",
    "details": null,
    "hint": null,
    "message": "invalid input syntax for type bigint: \"id_decode_once('jJPz')\""
}

obviously the decoding isn't taking place... is this even possible? Or do I need to then use postgres functions to ineract with any table where I transmitted an encoded it to the frontend?

**Comment by GaryAustin1** [comment, marked as answer]
You can't pass in functions to the rest API.
Normally I think you would put the hash id in a column and use that column in a filter.

**Reply by taeling** [reply]
Thanks for the confirmation, that's exactly what I had done in the meantime!  I added a generated column: `short_id text generated always as (id_encode (id)) stored`
